src/pybind/mgr/cephadm/services/osd.py

```diff
@@ -94,7 +98,9 @@ def deploy_osd_daemons_for_existing_osds(self, host: str, service_name: str,
                                              replace_osd_ids: Optional[List[str]] = None) -> str:
 
         if replace_osd_ids is None:
-            replace_osd_ids = self.find_destroyed_osds().get(host, [])
+            # Note: OSD id claims in CRUSH map are linked to the bare name of
+            # the hostname. Use bare name in case of FQDN hostnames.
+            replace_osd_ids = self.find_destroyed_osds().get(host.split(".")[0], [])
```

Comment:
Should this be done for previews as well? https://github.com/ceph/ceph/blob/master/src/pybind/mgr/cephadm/services/osd.py#L221

Reply:
Good catch Adam!!! Thank you man!!.